### DISCUSSION TITLE
isTouchSupported property is useless

### DISCUSSION THREAD

**Discussion by david-mark** [fabricjs/fabric.js#7971] [General]
This is useless:

`fabric.isTouchSupported = 'ontouchstart' in fabric.window || 'ontouchstart' in fabric.document ||
  (fabric.window && fabric.window.navigator && fabric.window.navigator.maxTouchPoints > 0);`

...and always has been, regardless of how many variations on this theme appear in libraries. There's just no useful inference to make from the result of the expression.

Kangax, if you are there, I'm sure you know this and know what to do instead, but am also fairly sure you have better things to do twelve years later. This thing is Kangax 2009, IIRC. :)

In any event, can somebody deprecate this? Any code that tries to derive meaning from this property is suspect.

**Comment by ShaMan123** [comment]
This library is actively maintained. Perhaps sometimes response takes time and things lag but fabric is alive.
You are more than welcome to PR. This is a huge code base and is undergoing a major refactoring.
Take part.

**Reply by ShaMan123** [reply]
We will probably deprecate the header file - the entire build process is going to be replaced with something modern and js syntax as well

**Comment by david-mark** [comment]
Sounds good. The thing is like a time capsule from 2009. :)

Now if I could just figure out why ever time I add text (or anything it seems) to a group, it always renders dead center. It's maddening, as the originX/Y properties seem to do nothing. I've seen your examples (more than I cared to see really) and they usually open by mutating the prototype of various objects to force "center" origin, but I'm not doing that in my script. And I've seen "fiddles" that seem to do what I'm trying to do, but their code looks basically like mine. And yes, using left and top 0. Tried "textAlign" too. BTW, the docs are bad too, listing properties that seemingly do nothing (e.g. backgroundColor for groups) and the explanations are generally deficient.

Initial impression: hate this thing, possibly even worse than D3, at least with respect to the groups feature. Nothing intuitive about it, and I'm not exactly new to this sort of thing. :(

On the PR, don't have time at the moment, but I imagine somebody else can remove the offending line.

**Reply by melchiar** [reply]
Your concerns are valid. The group system is currently undergoing an overhaul, as indicated in [this sticky tracker issue](https://github.com/fabricjs/fabric.js/issues/7670). Yes, the docs are also insufficient. There's been a lot of development recently in fabric and the docs haven't fully caught up to all of the recent changes. Doc improvements are on the roadmap and will get better in time. 

That being said, I have to wonder why you feel it necessary to degrade a project that's provided to you at no cost by a community of volunteers when you yourself aren't willing to contribute a PR to fix the line you find so offending.
